### PR TITLE
cascoda-sdk: Add package

### PR DESCRIPTION
**Pull request by elie-elkhoury**
**Maintainer:** me and @CiaranWoodward
**Compile tested:** arch: mips_24kc and mips_34kc, OpenWrt version: 15.05.1 "Chaos Calmer" and latest master.
**Run tested:** arch: mips_34kc, OpenWrt version: 15.05.1 "Chaos Calmer", tests done: ran the utilities to make sure they work as they're supposed to, including the utilities that are designed to test the interface.

**Description:**
This package installs some utilities for interacting with Cascoda IEEE 802.15.4 and Thread devices in various ways.
Includes a sniffer, serial adapter, network co-processor program, and some test utilities.

Signed-off-by: Elie El Khoury <e.khoury@cascoda.com>

### CODE REVIEW

**Comment by karlp** [comment]
Looks fine to me.  You may want to change some of those utility names though, they're incredibly general.

**Comment by elie-elkhoury** [comment]
> Looks fine to me. You may want to change some of those utility names though, they're incredibly general.

Thanks for the suggestion @karlp, I've done that now.

File: utils/cascoda-sdk/Makefile [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+endef
+
+define Package/cascoda-sdk/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/cliapp $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/sniffer $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/ncpapp $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/ot-server-standalone $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/serial-adapter $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/usb_test $(1)/usr/bin
```
Comment (karlp) [inline utils/cascoda-sdk/Makefile]: You changed the names in the description, but not what's actually installed?

Comment (elie-elkhoury) [reply utils/cascoda-sdk/Makefile]: Whoops, fixed it now.

File: utils/cascoda-sdk/Makefile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+endef
+
+define Package/cascoda-sdk/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/cliapp $(1)/usr/bin/ca-ot-cliapp
+	$(CP) $(PKG_BUILD_DIR)/bin/sniffer $(1)/usr/bin/ca-sniffer
+	$(CP) $(PKG_BUILD_DIR)/bin/ncpapp $(1)/usr/bin/ca-ot-ncpapp
+	$(CP) $(PKG_BUILD_DIR)/bin/ot-server-standalone $(1)/usr/bin/ca-ot-server-standalone
+	$(CP) $(PKG_BUILD_DIR)/bin/serial-adapter $(1)/usr/bin/ca-serial-adapter
+	$(CP) $(PKG_BUILD_DIR)/bin/usb_test $(1)/usr/bin/ca-comm-check
```
Comment (karlp) [inline utils/cascoda-sdk/Makefile]: Now, you just have to worry about users of openwrt having different commands than users of your original tooling :)  My suggestion was just a suggestion by the way, it isn't something you had to do to get the package merged.  If you want to revert this until/if you work out names on your own project, no problem.

Comment (elie-elkhoury) [reply utils/cascoda-sdk/Makefile]: We're happy with the names as they are right now, so we're fine with getting this merged as it is, unless someone else has any other suggestions :)

File: utils/cascoda-sdk/Makefile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+PKG_NAME:=cascoda-sdk
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/Cascoda/cascoda-sdk.git
+PKG_SOURCE_DATE:=2020-02-04
+PKG_SOURCE_VERSION:=745711e26b31c87d707ffc8054a94e3086be08f5
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_SOURCE_VERSION).tar.gz
```
Comment (neheb) [inline utils/cascoda-sdk/Makefile]: Three of these are default and not needed. MIRROR_HASH would need updating as default archive is .xz.

File: utils/cascoda-sdk/Makefile [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_SOURCE_VERSION)
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_SOURCE_VERSION)
+PKG_MIRROR_HASH:=64e27b514b7f07e3e8904170391d02aeb2fd561d043c40a853c4166928f28e09
+
+PKG_MAINTAINER:=Elie El Khoury <e.khoury@cascoda.com>, Ciaran Woodward <c.woodward@cascoda.com>
+
```
Comment (neheb) [inline utils/cascoda-sdk/Makefile]: Remove the empty line.

File: utils/cascoda-sdk/Makefile [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_CHECK_FORMAT_SECURITY:=0
```
Comment (neheb) [inline utils/cascoda-sdk/Makefile]: I don't like this. Is this needed?

File: utils/cascoda-sdk/Makefile [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/cascoda-sdk
+  SECTION:=cascoda-sdk
+  DEPENDS:=+libpthread +libstdcpp
```
Comment (neheb) [inline utils/cascoda-sdk/Makefile]: Can this work with uClibc++ or does this need C++11 and above?

File: utils/cascoda-sdk/Makefile [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+define Build/Configure
+  $(call Build/Configure/Default,--with-linux-headers=$(LINUX_DIR))
+	sed -i '/MBEDTLS_HAVE_ASM/d' $(PKG_BUILD_DIR)/_deps/openthread-src/third_party/mbedtls/mbedtls-config.h
+endef
+
+define Package/cascoda-sdk/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/cliapp $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/sniffer $(1)/usr/bin
+	$(CP) $(PKG_BUILD_DIR)/bin/ncpapp $(1)/usr/bin
```
Comment (neheb) [inline utils/cascoda-sdk/Makefile]: I would replace CP with INSTALL_BIN for all of these.

**Comment by elie-elkhoury** [comment]
Hosting package here:
https://github.com/Cascoda/cascoda-openwrt

**Pull request closed without merging**